### PR TITLE
[libsocialcache] Provide methods to remove OneDrive and Dropbox image…

### PR DESCRIPTION
**Pull request by antseppa**
…s from UI. Contributes to JB#32899

### CODE REVIEW

File: src/qml/dropbox/dropboximagecachemodel.cpp [hunk 1]
```diff
@@ -167,6 +167,42 @@ void DropboxImageCacheModel::setDownloader(DropboxImageDownloader *downloader)
...
+        if (image) {
+            DropboxAlbum::ConstPtr album = d->database.album(image->albumId());
+            if (album) {
+                d->database.addAlbum(album->albumId(), album->userId(), album->createdTime(),
+                                     album->updatedTime(), album->albumName(), album->imageCount()-1,
+                                     album->hash());
+            }
+        }
+
+        d->database.removeImage(imageId);
```
Comment (chriadam) [inline src/qml/dropbox/dropboximagecachemodel.cpp]: I assume that one also deletes the file from disk if it exists, not just the row from the database?

**Comment by chriadam** [comment]
LGTM

File: src/qml/dropbox/dropboximagecachemodel.cpp (hunk 1 above)
Comment (antseppa) [inline src/qml/dropbox/dropboximagecachemodel.cpp]: SocialImageCache will remove possible downloaded version after the cache period expires (usually two weeks).

Comment (chriadam) [inline src/qml/dropbox/dropboximagecachemodel.cpp]: Ah, of course.  I'd forgotten about that expiry stuff.  Thanks!

**Pull request merged**